component/telemetry.go

```diff
@@ -30,4 +32,7 @@ type TelemetrySettings struct {
 
 	// MeterProvider that the factory can pass to other instrumented third-party libraries.
 	MeterProvider metric.MeterProvider
+
+	// MetricsLevel controls the level of detail for metrics emitted by the collector.
+	MetricsLevel configtelemetry.Level
```

Comment:
Please add a comment that this is experimental. I am not yet convinced that we should expose this.

Reply:
What is our alternative when the flags are removed?  There is no longer a single point of truth for this information, which is in line with the desire to be able to have multiple functioning collector services in a single process.  Components need to know how the service that manages them was configured.

Reply:
Not sure, how does otel metrics will solve this? Most likely will be based on Views. Want to see how otel solves it because same problem applies there as well for any "instrumentation" correct?